HesoyamHospital/WebApplicationTests/Unit/Appointments/GetSuspiciousPatientsTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Backend.Model.UserModel;
+using Backend.Repository.Abstract.UsersAbstractRepository;
+using Moq;
+using Shouldly;
+using System.Collections.Generic;
+using Xunit;
+using WebApplication.Appointments.Service;
+using Backend.Repository.Abstract.MedicalAbstractRepository;
+using WebApplication.Appointments.DTOs;
+using Backend.Model.PatientModel;
+
+namespace WebApplicationTests.Unit.Appointments
+{
+    public class GetSuspiciousPatientsTests
+    {
+
+        [Fact]
+        public void Get_accurate_count_of_suspicious_patients()
+        {
+            AppointmentService service = new AppointmentService(CreateStubRepository(), CreateAppointmentStubRepository(), CreateCancellationStubRepository());
+
+            List<BlockPatientDTO> blockedPatients = service.GetSuspiciousPatients();
+
+            blockedPatients.Count.ShouldBeEquivalentTo(1);
+        }
+
+        [Fact]
+        public void Get_suspicious_patients()
+        {
+            AppointmentService service = new AppointmentService(CreateStubRepository(), CreateAppointmentStubRepository(), CreateCancellationStubRepository());
+
+            List<BlockPatientDTO> blockedPatients = service.GetSuspiciousPatients();
+
+            blockedPatients[0].Username.ShouldBe("perapera");
+        }
+
+
+        private static IPatientRepository CreateStubRepository()
+        {
+            var stubRepository = new Mock<IPatientRepository>();
+            Patient patient1 = new Patient(0);
+            patient1.Appointments = new List<Appointment>();
```

Comment:
Just a suggestion, use constructor instead.

Reply:
Thank you for the suggestion, I will pay more attention from now on.